.github/workflows/sonarqube.yml

```diff
@@ -0,0 +1,81 @@
+name: Sonarqube nightly
+
+on:
+  schedule:
+    - cron: '0 20 * * *'
+
+# Enrich gradle.properties for CI/CD
+env:
+  CI_GRADLE_ARG_PROPERTIES: >
+    -Porg.gradle.jvmargs=-Xmx4g
+    -Porg.gradle.parallel=false
+jobs:
+  codecov-units:
+    name: Unit tests with code coverage
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-java@v3
+        with: 
+          distribution: 'adopt'
+          java-version: '11'
+      - uses: actions/cache@v3
+        with:
+          path: |
+            ~/.gradle/caches
+            ~/.gradle/wrapper
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
+      - run: ./gradlew allCodeCoverageReport $CI_GRADLE_ARG_PROPERTIES
+      - name: Upload Codecov data
+        uses: actions/upload-artifact@v3
+        if: always()
+        with:
+          name: codecov-xml
+          path: |
+            build/reports/jacoco/allCodeCoverageReport/allCodeCoverageReport.xml
+
+  sonarqube:
+    name: Sonarqube upload
+    runs-on: ubuntu-latest
+    needs:
+      - codecov-units
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-java@v3
+        with:
+          distribution: 'adopt'
+          java-version: '11'
+      - uses: actions/cache@v3
+        with:
+          path: |
+            ~/.gradle/caches
+            ~/.gradle/wrapper
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
+      - uses: actions/download-artifact@v3
+        with:
+          name: codecov-xml # will restore to allCodeCoverageReport.xml by default; we restore to the same location in following tasks
+      - run: mkdir -p build/reports/jacoco/allCodeCoverageReport/
+      - run: mv allCodeCoverageReport.xml build/reports/jacoco/allCodeCoverageReport/
+      - run: ./gradlew sonarqube $CI_GRADLE_ARG_PROPERTIES
+        env:
+          ORG_GRADLE_PROJECT_SONAR_LOGIN: ${{ secrets.SONAR_TOKEN }}
+
+# Notify the channel about sonarqube failures
+  notify:
+    name: Notify matrix
+    runs-on: ubuntu-latest
+    needs:
+      - sonarqube
+      - codecov-units
+    if: always() && needs.sonarqube.result != "success" and needs.codecov-units.result != "success"
```

Comment:
For this condition, I would have expect something like: 

`if: always() && (needs.sonarqube.result != "success" || needs.codecov-units.result != "success")`

I guess we should notify in case either the sonarqube job or the code coverage job fails. And also I am not sure we can use `and` instead of `&&`? What do you think?

Reply:
Argh, yes; I made this fix in post-pr.yml but not here - good catch.